apps/system/js/screen_manager.js

```diff
@@ -292,11 +319,15 @@ var ScreenManager = {
         break;
 
       case 'callschanged':
-        if (!telephony.calls.length &&
-            !(telephony.conferenceGroup &&
-              telephony.conferenceGroup.calls.length)) {
-
-          this.turnScreenOn();
+        if (numOpenLines() === 0) {
+          if (this._turnOffWhenCallsEnd) {
+            this._turnOffWhenCallsEnd = false;
+            setTimeout(function() {
+              this.turnScreenOff(/* instant */ true, 'powerkey');
+            }.bind(this), 2000);
```

Comment:
See above, this is because the callscreen is hidden after 2s when hanging up the last call.